dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -151,4 +205,14 @@ def allows_silent_takeover(oauth_user)
     lookup_user = User.find_by_email_or_hashed_email(oauth_user.email)
     allow_takeover && lookup_user
   end
+
+  def should_connect_provider?
+    return current_user && session[:connect_provider].present?
+  end
+
+  def can_connect_provider?
+    is_valid_connection = session[:connect_provider]&.future?
+    session[:connect_provider] = nil
+    return is_valid_connection
```

Comment:
Style nit: Redundant return, and might be expressed more concisely thus?
```rb
connect_flag_expiration = session.delete :connect_provider
connect_flag_expiration&.future?
```
Pretty sure [`Rack::Session::Abstract::SessionHash#delete`](https://www.rubydoc.info/gems/rack/Rack/Session/Abstract/SessionHash#delete-instance_method) for rack 2.0.5 applies here.